lib/schedule/data.ex

```diff
@@ -441,6 +588,19 @@ defmodule Schedule.Data do
     }
   end
 
+  @spec active_services_on_dates([Date.t()], tables()) :: %{Date.t() => [Service.id()]}
+  defp active_services_on_dates(dates, %{calendar: calendar_table}) do
+    dates
+    |> Enum.map(fn date ->
```

Comment:
suggestion: Could use `:mnesia.select` to query all the dates at once, and then something like `Enum.group_by` to turn them into a map.